contrib/apelscripts/condor_batch_blah.py

```diff
@@ -130,6 +135,13 @@ def condor_q_format(job_history: Path, *formats: str) -> str:
 else:
     batch_stream = blah_stream = sys.stdout
 
+accounting_user_expr = "ifThenElse(!isUndefined(x509userproxysubject), x509userproxysubject, orig_AuthTokenSubject)"
+accounting_group_expr = "ifThenElse(!isUndefined(x509UserProxyFirstFQAN), x509UserProxyFirstFQAN, %s)"
+if use_wlcg_groups:
+    accounting_group_expr = accounting_group_expr % 'ifThenElse(!isUndefined(orig_AuthTokenGroups), split(orig_AuthTokenGroups, ",")[0], %s)'
+accounting_group_expr = accounting_group_expr % 'ifThenElse(!isUndefined(userMap("ApelAGMap", Owner)), userMap("ApelAGMap", Owner), %s)'
+accounting_group_expr = accounting_group_expr % 'ifThenElse(!isUndefined(orig_AuthTokenIssuer) && !isUndefined(orig_AuthTokenSubject) && !isUndefined(userMap("ApelAGMap", strcat(orig_AuthTokenIssuer, ",", orig_AuthTokenSubject))), userMap("ApelAGMap", strcat(orig_AuthTokenIssuer, ",", orig_AuthTokenSubject)), %s)'
```

Comment:
Similar to the above, you should be able to save the double `userMap` here too though it isn't quite as nice.
```suggestion
accounting_group_expr = accounting_group_expr % '(((orig_AuthTokenIssuer isnt UNDEFINED && orig_AuthTokenSubject isnt UNDEFINED) ? userMap("ApelAGMap", strcat(orig_AuthTokenIssuer, ",", orig_AuthTokenSubject)) : UNDEFINED) ?: %s)'
```

Reply:
I do not see why there need to be "(((" and "))" instead of "((" and ")" ?
Also, is the ": UNDEFINED" really needed?

Reply:
I'm not entirely sure about the precedence rule of ClassAd operators, so the extra parentheses are to be conservative. I guess the pair around the first condition could be removed.

The `: UNDEFINED` is needed to have the full ternary – something must be produced in that branch, and the next ladder step expects a result or `UNDEFINED`. Sadly, [`strcat` evaluates to `ERROR` on `UNDEFINED` input](https://htcondor.readthedocs.io/en/latest/classads/classad-builtin-functions.html#strcat()) so one can't just use the `?:` operator as before.

FWIW, since the next step is a constant `UNDEFINED` anyways, one could skip the logic for inserting a placeholder branch and just stop at `UNDEFINED`. It would also probably look a bit nicer if the expression were built in reverse (as I did in `format_apel_scaling`) which allows to automatically create the scaffolding for the ladder, without hardcoding it into every step.